From 427d4c1e5691cbf855371d6a0e30dd1f67726e5e Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Thu, 27 Aug 2015 13:59:10 +0800
Subject: [PATCH 0338/5242] MLK-11429-26: ASoC: fsl: imx-si476x: tdm_slot is
 changed in fsl_ssi

commit  8a2da8bdf0a641f0cedb95956b8ee7635661d3e0 from
https://source.codeaurora.org/external/imx/linux-imx.git

The fsl_ssi revert the mask in tdm_slot, so machine use 1 present enable
not 0.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-si476x.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/fsl/imx-si476x.c b/sound/soc/fsl/imx-si476x.c
index 276ae46..ea9edba 100644
--- a/sound/soc/fsl/imx-si476x.c
+++ b/sound/soc/fsl/imx-si476x.c
@@ -68,8 +68,8 @@ static int imx_si476x_hw_params(struct snd_pcm_substream *substream,
 	}
 
 	ret = snd_soc_dai_set_tdm_slot(cpu_dai,
-			channels == 1 ? 0xfffffffe : 0xfffffffc,
-			channels == 1 ? 0xfffffffe : 0xfffffffc,
+			channels == 1 ? 1 : 0x3,
+			channels == 1 ? 1 : 0x3,
 			2, 32);
 	if (ret) {
 		dev_err(cpu_dai->dev, "failed to set dai tdm slot\n");
-- 
1.7.9.5

